apps/gallery/js/ImageEditor.js

```diff
@@ -1221,8 +1229,51 @@ ImageEditor.prototype.getFullSizeBlob = function(type, done, progress) {
       setTimeout(function() {
         done(blob);
       });
+    }
+
+    function updateExif(blob, metaData, callback) {
+      metaData.Orientation = 1;
+      if (metaData.PixelXDimension) {
+        metaData.PixelXDimension = canvas.width;
+      }
+      if (metaData.PixelYDimension) {
+        metaData.PixelYDimension = canvas.height;
+      }
+      JPEGParser.writeExifMetaData(
+        blob, metaData,
+        (error, modifiedBlob) => {
+          if (error) {
+            console.error('Error' + error);
+          }
+          // Process modified file
+          callback(modifiedBlob);
+        });
+    }
+
+    var originalBlob = this.imageBlob;
+    var originalExif = this.imageExif;
```

Comment:
If you use this.edits.EXIF here instead of this.imageExif, you might actually have a value ready for use.